src/main/scala/com/microsoft/azure/iot/iothubreact/checkpointing/Backends/CheckpointBackend.scala

```diff
@@ -6,7 +6,9 @@ import com.microsoft.azure.iot.iothubreact.checkpointing.Configuration
 
 trait CheckpointBackend {
 
-  def checkpointNamespace: String = Configuration.storageNamespace
+  val cfg: Configuration
+
+  def checkpointNamespace: String = cfg.storageNamespace
 
```

Comment:
this way configuration is added to the trait which is a bit out of scope, and the overlapping name with the var used in the AzureBlob and Cassandra classes leads to some ambiguity. 
I would suggest instead to use an implicit, which is consistent with the other places:

```
trait CheckpointBackend {

  def checkpointNamespace(implicit config: ICPConfiguration): String = config.storageNamespace
```